src/solver/variable/economy/links/ntc.h

```diff
@@ -0,0 +1,344 @@
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+#ifndef __SOLVER_VARIABLE_ECONOMY_NTC_H__
+#define __SOLVER_VARIABLE_ECONOMY_NTC_H__
+
+#include "../../variable.h"
+#include <antares/study/area/constants.h>
+
+namespace Antares
+{
+namespace Solver
+{
+namespace Variable
+{
+namespace Economy
+{
+/*
+ NTC capacities.
+ Note that no columns are produced (categoryFileLevel = 0)
+ These data exist for the sake of quadratic optimization (FLOW QUAD variable, see flowQuad.h),
+ for which we need the average houly direct/indirect NTC for each link.
+*/
+struct VCardNTC
+{
+    //! Caption
+    static const char* Caption()
+    {
+        return "NTC";
+    }
+    //! Unit
+    static const char* Unit()
+    {
+        return "MWh";
+    }
+
+    //! The short description of the variable
+    static const char* Description()
+    {
+        return "Net transfer capacities";
+    }
+
+    //! The expecte results
+    typedef Results<R::AllYears::Average<>> // The average values throughout all years
+      ResultsType;
+
+    enum
+    {
+        //! Data Level
+        categoryDataLevel = Category::link,
+        //! File level (provided by the type of the results)
+        categoryFileLevel = 0, // No output is produced, this is just a temporary variable
+        //! Precision (views)
+        precision = Category::hourly,
+        //! Indentation (GUI)
+        nodeDepthForGUI = +0,
+        //! Decimal precision
+        decimal = 0,
+        //! Number of columns used by the variable (One ResultsType per column)
+        columnCount = 2,
+        //! The Spatial aggregation
+        spatialAggregate = Category::spatialAggregateSum,
+        spatialAggregateMode = Category::spatialAggregateEachYear,
+        spatialAggregatePostProcessing = 0,
+        //! Intermediate values
+        hasIntermediateValues = 0,
+        //! Can this variable be non applicable (0 : no, 1 : yes)
+        isPossiblyNonApplicable = 1,
+    };
+
+    typedef IntermediateValues IntermediateValuesBaseType[columnCount];
+    typedef IntermediateValuesBaseType* IntermediateValuesType;
+
+    struct Multiple
+    {
+        static const char* Caption(uint indx)
+        {
+            switch (indx)
+            {
+            case 0:
+                return "NTC direct";
+            case 1:
+                return "NTC indirect";
+            default:
+                return "<unknown>";
+            }
+        }
+    };
+
+}; // class VCard
+
+/*!
+** \brief NTC
+*/
+template<class NextT = Container::EndOfList>
+class NTC : public Variable::IVariable<NTC<NextT>, NextT, VCardNTC>
```

Comment:
I am not a fan of adding this new output variable in order to make some average computations for the quadratique optimization. For me, it's a kind of over-engineering and, as this variable is not a real output variable (it's not printed as a result), it can be confusing. I understand that this way of doing things is quite easy, and was made on the model of what is currently made for **balance** regarding the quadratic optimization. But still, there should be another way of making things.
By the way, this variable appears in the thematic trimming tab, and we don't want that, I think.

Reply:
I no longer use a new output variable.